src/core/src/any.cpp

```diff
@@ -96,7 +96,7 @@ void Any::read(std::istream& istream) {
 }
 
 bool Any::operator==(const Any& other) const {
-    if (_impl == nullptr && other._impl == nullptr) {
+    if (_impl == nullptr || other._impl == nullptr) {
```

Comment:
CC @apankratovantonp

Reply:
if any of the _impl is not nullptr but other is nullptr it is covered by next check.

Reply:
I am not sure that it is correct check.
This code is incorrect for nullptr: `*other._impl` because it calls dereferencing of nullptr (undefined behavior)

Reply:
got it